### PR TITLE
Refactor  getLogicAndActualTables

### LINKED ISSUE #19326: Refactor TableToken
## Feature Request

 Uniform `getLogicAndActualTables()`  of TableToken, CursorToken, IndexToken, ConstraintToken

I try to work it when the test(https://github.com/apache/shardingsphere/pull/19325) be merged.

### PR DESCRIPTION
**Pull request by sunkai-cai**
Fixes #19326.

Changes proposed in this pull request:
-  Uniform getLogicAndActualTables() of TableToken, CursorToken, IndexToken, ConstraintToken
-  Refactor  TableToken.constructor

### CODE REVIEW

File: shardingsphere-features/shardingsphere-sharding/shardingsphere-sharding-core/src/main/java/org/apache/shardingsphere/sharding/rewrite/token/pojo/TokenUtil.java [hunk 1]
```diff
@@ -0,0 +1,52 @@
...
+import org.apache.shardingsphere.infra.binder.statement.SQLStatementContext;
+import org.apache.shardingsphere.infra.route.context.RouteMapper;
+import org.apache.shardingsphere.infra.route.context.RouteUnit;
+import org.apache.shardingsphere.sharding.rule.ShardingRule;
+
+/**
+ * Token utility.
+ */
+@NoArgsConstructor(access = AccessLevel.PRIVATE)
+public final class TokenUtil {
```
[the remark this answers is not in the document]
Comment (sunkai-cai) [reply shardingsphere-features/shardingsphere-sharding/shardingsphere-sharding-core/src/main/java/org/apache/shardingsphere/sharding/rewrite/token/pojo/TokenUtil.java:34]: I try it

File: shardingsphere-features/shardingsphere-sharding/shardingsphere-sharding-core/src/main/java/org/apache/shardingsphere/sharding/rewrite/token/pojo/TokenUtil.java [hunk 2]
```diff
@@ -0,0 +1,52 @@
...
+    
+    /**
+     * get logicTable and actualTable.
```
[the remark this answers is not in the document]
Comment (sunkai-cai) [reply shardingsphere-features/shardingsphere-sharding/shardingsphere-sharding-core/src/main/java/org/apache/shardingsphere/sharding/rewrite/token/pojo/TokenUtil.java:37]: ok

File: shardingsphere-features/shardingsphere-sharding/shardingsphere-sharding-core/src/main/java/org/apache/shardingsphere/sharding/rule/ShardingRule.java [hunk 3]
```diff
@@ -670,6 +672,22 @@ public Map<String, String> getLogicAndActualTablesFromBindingTable(final String
...
+     *
+     * @param routeUnit Route unit
+     * @param availableLogicBindingTables available logic binding table names
+     * @return key is logicTable name, values is actualTable belong to this data source
+     */
+    public Map<String, String> getLogicAndActualTablesFromRouteUnit(final RouteUnit routeUnit, final Collection<String> availableLogicBindingTables) {
+        Map<String, String> result = new HashMap<>(availableLogicBindingTables.size(), 1);
+        for (RouteMapper each : routeUnit.getTableMappers()) {
+            result.put(each.getLogicName().toLowerCase(), each.getActualName());
+            result.putAll(this.getLogicAndActualTablesFromBindingTable(routeUnit.getDataSourceMapper().getLogicName(), each.getLogicName(), each.getActualName(), availableLogicBindingTables));
```
Comment (sunkai-cai) [inline shardingsphere-features/shardingsphere-sharding/shardingsphere-sharding-core/src/main/java/org/apache/shardingsphere/sharding/rule/ShardingRule.java:686]: I don't know why is do that.  Get last actualTableName from bindingTable ?

Comment (terrymanu) [reply shardingsphere-features/shardingsphere-sharding/shardingsphere-sharding-core/src/main/java/org/apache/shardingsphere/sharding/rule/ShardingRule.java:686]: Where is the code come from?

Comment (terrymanu) [reply shardingsphere-features/shardingsphere-sharding/shardingsphere-sharding-core/src/main/java/org/apache/shardingsphere/sharding/rule/ShardingRule.java:686]: What is the relationship about this method and ShardingRule? Why put the method here?

Comment (sunkai-cai) [reply shardingsphere-features/shardingsphere-sharding/shardingsphere-sharding-core/src/main/java/org/apache/shardingsphere/sharding/rule/ShardingRule.java:686]: https://github.com/apache/shardingsphere/blob/d0ca7711d4050f2d028d97538dfbec4f7cb50ccb/shardingsphere-features/shardingsphere-sharding/shardingsphere-sharding-core/src/main/java/org/apache/shardingsphere/sharding/rewrite/token/pojo/TableToken.java

Comment (sunkai-cai) [reply shardingsphere-features/shardingsphere-sharding/shardingsphere-sharding-core/src/main/java/org/apache/shardingsphere/sharding/rule/ShardingRule.java:686]: I just remove `getLogicAndActualTables` to ShardingRule

Comment (sunkai-cai) [reply shardingsphere-features/shardingsphere-sharding/shardingsphere-sharding-core/src/main/java/org/apache/shardingsphere/sharding/rule/ShardingRule.java:686]: > What is the relationship about this method and ShardingRule? Why put the method here?

yes, it's probably not a good idea.

**Comment by sunkai-cai** [comment]
I have not a good idea of TokenUtil class name.  maybe we can refactor it  in the next time.

**Comment by terrymanu** [comment]
> I have not a good idea of TokenUtil class name. maybe we can refactor it in the next time.

ok, just revise other code review suggestion please.

**Review by terrymanu — APPROVED** [review]

**Pull request merged**